packages/backend-defaults/src/entrypoints/database/connectors/postgres.ts

```diff
@@ -64,18 +64,45 @@ export function createPgDatabaseClient(
  * @param dbConfig - The database config
  * @param overrides - Additional options to merge with the config
  */
-export function buildPgDatabaseConfig(
+export async function buildPgDatabaseConfig(
   dbConfig: Config,
   overrides?: Knex.Config,
 ) {
-  return mergeDatabaseConfig(
+  const config = mergeDatabaseConfig(
     dbConfig.get(),
     {
       connection: getPgConnectionConfig(dbConfig, !!overrides),
       useNullAsDefault: true,
     },
     overrides,
   );
+
+  if (config.client === 'pg+google-cloudsql') {
+    const {
+      Connector: CloudSqlConnector,
+      IpAddressTypes,
+      AuthTypes,
+    } = await import('@google-cloud/cloud-sql-connector');
+    // override the config to be pg for backwards compat with other code
+    config.client = 'pg';
```

Comment:
is the result of `config.get()` safely mutable? 🤔 Maybe you could add `{}` as the first argument to `mergeDatabaseConfig` to make sure that it's a separate object?